pytorch_lightning/trainer/trainer.py

```diff
@@ -856,11 +922,18 @@ def tune(
 
     def call_setup_hook(self, model):
         # call setup after the ddp process has connected
-        stage_name = 'test' if self.testing else 'fit'
+        stage_name = self.evaluating or 'fit'
+
         if self.datamodule is not None:
-            called = self.datamodule.has_setup_test if self.testing else self.datamodule.has_setup_fit
+            called = {
+                None: self.datamodule.has_setup_fit,
```

Comment:
Could use the enum there. It is weird to have None as a key.

Reply:
Now that I look at it again, it's weird that I didn't use `stage_name` directly to select a key, it would've been much clearer in that case:

```
            called = {
                'fit': self.datamodule.has_setup_fit,
                'validation': self.datamodule.has_setup_validation,
                 'test': self.datamodule.has_setup_test,
             }[stage_name]
```

Regarding the enum, it seems a very good idea. I think the real benefits, though, would come from also using it when calling `self.datamodule.setup(stage_name)` and `model.setup(stage_name)`, instead of a plain string. That way, it's explicit to the users what are the stages that they need to handle in the setup method. Unfortunately, it's a much bigger refactor that way, which feels a bit out-of-scope for this PR.

Reply:
In the meantime, I updated the lookup to use `stage_name` instead of `self.evaulating` (see 595f4e8). For the enum discussion, I'd continue on the thread above with you and @carmocca.